modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/generalized_plane_strain.md

```diff
@@ -0,0 +1,125 @@
+# Generalized Plane Strain
+
+## Description
+
+The generalized plane strain problem has found use in many important applications. Bearing the same name as generalized plane strain problem, different presentations have been found in the literature. The simplest one is an extension of the plane strain problem by allowing a non-vanishing constant direction strain in the out-of-plane direction [citep:Adams1967gps]. This has been further generalized by allowing two rotations degrees of freedom [citep:Abaqus2014gps]. An even more generalized form includes the anticlastic problem associated with the out-of-plane shear [citep:Adams1984gps, Li1999gps]. In the [tensor mechanics module](tensor_mechanics/index.md), the simplest form was implemented, i.e. an extra degree of freedom representing the out-of-plane direct strain is included in addition to the conventional plane strain problem [citep:Li2005gps].
+
+## Formulation
+
+For the description purpose, it is convenient to introduce the coordinate system so that the $x$-$y$ plane is positioned transverse to the infinite out-of-plane direction and the $z$-axis runs in the out-of-plane direction. Under the simplest generalized plane strain conditions, the in-plane stresses are not functions of $z$, implying that the in-plane strains are not functions of $z$ from the constitutive relationship for linearly elastic and homogeneous materials. The generalized plane strain problem is 2D in presentation, defined in a 2D domain.
```

Comment:
How about starting this paragraph with "We introduce the coordinate system ...."?

Reply:
And make the last sentence bold so that it jumps out at the reader